pkg/prometheus/operator.go

```diff
@@ -229,6 +229,16 @@ func (sr *StatusReporter) Process(ctx context.Context, p monitoringv1.Prometheus
 		}
 	}
 
+	// Indicate that no scrape resource selectors are specified. Do not take precedence over an unhealthy condition.
+	allSelectorsNil := commonFields.ServiceMonitorSelector == nil &&
+		commonFields.PodMonitorSelector == nil &&
+		commonFields.ProbeSelector == nil &&
+		commonFields.ScrapeConfigSelector == nil
+	gotCondition := sr.Reconciliations.GetCondition(key, p.GetObjectMeta().GetGeneration())
+	if allSelectorsNil && gotCondition.Status == monitoringv1.ConditionTrue {
+		gotCondition.Reason = "NoScrapeResourceSelected"
+		gotCondition.Message = "None of the scrape resource selectors are specified."
```

Comment:
```suggestion
		gotCondition.Message = "The operator selected no ServiceMonitor, PodMonitor, Probe and ScrapeConfig resources because all resource selectors are null."
```